### ISSUE TITLE
[Bug] Update current_rl XTTS

### ISSUE THREAD

**Issue by insomnia777** [coqui-ai/TTS#3483]
### Describe the bug

Hi!

Is the learning rate update actually happening?

![2024-01-01-113616_640x162_scrot](https://github.com/coqui-ai/TTS/assets/88548335/c757f546-6b62-4bae-9375-e5babb338ba0)
![2024-01-01-113625_630x153_scrot](https://github.com/coqui-ai/TTS/assets/88548335/361736c5-3c98-414a-b28e-c124b519e0df)

Is the learning rate update actually happening? After 10,000 steps it should be 0.004, but I constantly see the initial learning rate. It's annoying.

### To Reproduce

def main():
   
    model_args = GPTArgs(
        max_conditioning_length=132300,
        min_conditioning_length=66150,
        debug_loading_failures=False,
        max_wav_length=384000,
        max_text_length=180,
        mel_norm_file=MEL_NORM_FILE,
        dvae_checkpoint=DVAE_CHECKPOINT,
        xtts_checkpoint=XTTS_CHECKPOINT, 
        tokenizer_file=TOKENIZER_FILE,
        gpt_num_audio_tokens=1026,
        gpt_start_audio_token=1024,
        gpt_stop_audio_token=1025,
        gpt_use_masking_gt_prompt_approach=True,
        gpt_use_perceiver_resampler=True,
    )
    
    audio_config = XttsAudioConfig(sample_rate=22050, dvae_sample_rate=22050, output_sample_rate=24000)
    
    config = GPTTrainerConfig(
        epochs=1000,
        output_path=OUT_PATH,
        model_args=model_args,
        run_name=RUN_NAME,
        project_name=PROJECT_NAME,
        run_description="""
            GPT XTTS training
            """,
        dashboard_logger=DASHBOARD_LOGGER,
        logger_uri=LOGGER_URI,
        audio=audio_config,
        batch_size=BATCH_SIZE,
        batch_group_size=48,
        eval_batch_size=BATCH_SIZE,
        num_loader_workers=16,
        eval_split_max_size=512,
        print_step=1130,
        plot_step=1130,
        log_model_step=1000,
        save_step=100000,
        save_n_checkpoints=1,
        save_checkpoints=True,
        #text_cleaner="phoneme_cleaners",
        use_phonemes=True,
        phoneme_language="ru-ru",
        #mixed_precision=True,
        # target_loss="loss",
        print_eval=False,
        optimizer="AdamW",
        optimizer_wd_only_on_weights=OPTIMIZER_WD_ONLY_ON_WEIGHTS,
        optimizer_params={"betas": [0.9, 0.96], "eps": 1e-8, "weight_decay": 1e-2},
        lr=5e-04,
        lr_scheduler="StepwiseGradualLR",
        lr_scheduler_params={
            "gradual_learning_rates": [
                [0, 5e-4],
                [10000, 4e-4],
                [20000, 3e-4],
                [30000, 2e-4],
                [40000, 1e-4],
                [50000, 9e-5],
                [60000, 8e-5],
                [70000, 7e-5],
                [80000, 6e-5]
            ]
        },
        scheduler_after_epoch=True,
        test_sentences=[
            {
                "text": "На стуле у оловянного тазика висела куртка Тамары, две узкие кровати были сдвинуты, шерстяные одеяла смяты, окно оставили открытым, и через него проникало осеннее солнце.",
                "speaker_wav": SPEAKER_REFERENCE,
                "language": LANGUAGE,
            },
        ],
    )

    # init the model from config
    model = GPTTrainer.init_from_config(config)

    # load training samples
    train_samples, eval_samples = load_tts_samples(
        DATASETS_CONFIG_LIST,
        eval_split=True,
        eval_split_max_size=config.eval_split_max_size,
        eval_split_size=config.eval_split_size,
    )
    import torch
    torch.cuda.set_device(0)
    # init the trainer and
    trainer = Trainer(
        TrainerArgs(
            restore_path=None,  # xtts checkpoint is restored via xtts_checkpoint key so no need of restore it using Trainer restore_path parameter
            skip_train_epoch=False,
            start_with_eval=START_WITH_EVAL,
            grad_accum_steps=GRAD_ACUMM_STEPS,
        ),
        config,
        output_path=OUT_PATH,
        model=model,
        train_samples=train_samples,
        eval_samples=eval_samples,
    )
    trainer.fit()

### Expected behavior

Correct learning rate

### Environment

```shell
{
    "CUDA": {
        "GPU": [
            "NVIDIA GeForce RTX 4090",
            "NVIDIA GeForce GTX 1060 6GB",
            "NVIDIA GeForce GTX 1060 6GB"
        ],
        "available": true,
        "version": "12.1"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.1.2",
        "TTS": "0.22.0",
        "numpy": "1.26.2"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            "ELF"
        ],
        "processor": "x86_64",
        "python": "3.11.7",
        "version": "#101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023"
    }
}
```

**Comment by MiXaiLL76** [comment]
use this 
```python3
lr_scheduler_params={
    "milestones": [int(num_epochs * 0.6), int(num_epochs * 0.8)],
    "gamma": 0.75,
    "last_epoch": -1,
},
scheduler_after_epoch=True,
```
